tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_service.py

```diff
@@ -62,32 +55,68 @@ def move_bucket(config, parsed_args, cloud_logger):
         conf=parsed_args, source_bucket=source_bucket)
 
     _check_bucket_lock(cloud_logger, config, source_bucket)
-    target_temp_bucket = _create_temp_target_bucket(cloud_logger, config,
-                                                    source_bucket_details)
 
-    # Create STS client
     sts_client = discovery.build(
         'storagetransfer', 'v1', credentials=config.target_project_credentials)
 
-    sts_account_email = _assign_sts_permissions(cloud_logger, sts_client,
-                                                config, target_temp_bucket)
-    _run_and_wait_for_sts_job(sts_client, config.target_project,
-                              config.bucket_name, config.temp_bucket_name,
-                              cloud_logger)
-
-    _delete_empty_source_bucket(cloud_logger, source_bucket)
-    _recreate_source_bucket(cloud_logger, config, source_bucket_details)
-    _assign_sts_permissions_to_new_bucket(cloud_logger, sts_account_email,
-                                          config)
-    _run_and_wait_for_sts_job(sts_client, config.target_project,
-                              config.temp_bucket_name, config.bucket_name,
-                              cloud_logger)
-
-    _delete_empty_temp_bucket(cloud_logger, target_temp_bucket)
-    _remove_sts_permissions(cloud_logger, sts_account_email, config)
+    if config.is_rename:
+        target_bucket = _create_target_bucket(cloud_logger, config,
```

Comment:
As discussed offline, it would help with maintainability and readability to decompose this into separate `move_bucket()` and `rename_bucket()` helper methods, and change the current method to `main()`.

Reply:
Refactored into smaller, more logical methods.